auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -587,6 +519,120 @@ public MethodSpec createReadMethod(ClassName className,
     return readMethod.build();
   }
 
+  private MethodSpec createReadMethod(ClassName className,
+      TypeName autoValueClassName,
+      List<Property> properties,
+      ImmutableMap<Property, FieldSpec> adapters,
+      TypeElement autoValueClassElement,
+      BuilderContext builderContext,
+      ProcessingEnvironment processingEnvironment) {
+    ParameterSpec jsonReader = ParameterSpec.builder(JsonReader.class, "jsonReader").build();
+    MethodSpec.Builder readMethod = MethodSpec.methodBuilder("read")
+        .addAnnotation(Override.class)
+        .addModifiers(PUBLIC)
+        .returns(autoValueClassName)
+        .addParameter(jsonReader)
+        .addException(IOException.class);
+
+    ClassName token = ClassName.get(JsonToken.NULL.getDeclaringClass());
+
+    readMethod.beginControlFlow("if ($N.peek() == $T.NULL)", jsonReader, token);
+    readMethod.addStatement("$N.nextNull()", jsonReader);
+    readMethod.addStatement("return null");
+    readMethod.endControlFlow();
+
+    Optional<ExecutableElement> defaultBuilder = findDefaultBuilder(autoValueClassElement, builderContext.builderClass());
+
+    ClassName builderClassName = ClassName.get(builderContext.builderClass());
+    FieldSpec builder = FieldSpec.builder(builderClassName, "_builder").build();
+    if (defaultBuilder.isPresent()) {
+      ClassName baseClass = ClassName.get(autoValueClassElement);
```

Comment:
We need it to pick out the build method in the builder. See the call to `findBuildMethod` near the end of this method.